turn_alerts/views.py

```diff
@@ -0,0 +1,128 @@
+
+        on_fallback_channel = request.headers.get("X-Turn-Fallback-Channel", "0") == "1"
+        is_turn_event = request.headers.get("X-Turn-Event", "0") == "1"
+
+        for alert in turn_alerts:
+            journey_id = alert.journey_id
+            org_error_code = alert.error_code
+            engage_token = organization.engage_token
+            engage_url = organization.engage_url
+
+        if webhook_type == "whatsapp" or is_turn_event:
+            WhatsAppWebhookSerializer(data=request.data).is_valid(raise_exception=True)
+            for inbound in request.data.get("messages", []):
+
+                message_type = inbound.pop("type")
+                direction = inbound["_vnd"]["v1"]["direction"]
+                message_requests_total.labels(
+                    fallback_channel=on_fallback_channel,
+                    direction=direction,
+                    message_type=message_type,
+                ).inc()
+            for statuses in request.data.get("statuses", []):
+                message_status = statuses.get("status")
+
+                if "errors" in statuses:
+                    recipient_id = statuses.get("recipient_id")
+                    error_code = statuses["errors"][0].get("code")
+                    event_count.labels(
+                        message_status=message_status,
+                        error_code=error_code,
+                        conversation_id=None,
+                        conversation_type=None,
+                    ).inc()
+                    if error_code == org_error_code:
+                        match = re.match(
+                            (
+                                r"^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]"
+                                r"*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$"
+                            ),
+                            recipient_id,
+                        )
+                        if match:
+                            start_turn_journey(
+                                recipient_id, journey_id, engage_url, engage_token
+                            )
+
+                else:
+                    recipient_id = statuses.get("recipient_id")
+                    conversation_id = statuses.get("conversation").get("id")
+                    conversation_type = (
+                        statuses.get("conversation").get("origin").get("type")
+                    )
+                    event_count.labels(
+                        message_status=message_status,
+                        error_code=None,
+                        conversation_id=conversation_id,
+                        conversation_type=conversation_type,
+                    ).inc()
+
+        elif webhook_type == "turn":
+            TurnOutboundSerializer(data=request.data).is_valid(raise_exception=True)
+            outbound = request.data
+            direction = outbound["_vnd"]["v1"].get("direction")
```

Comment:
Is this sometimes not available? I would assume its always there, message can only be inbound or outbound

Reply:
I haven't seen it not available but I just used a get in case it happens albeit quite unlikely.